### PR TITLE
Integrate changes from d2-reports

### PR DESCRIPTION
**Pull request by tokland**
- Add changes from d2-reports
- Fix tests
- Add script example
- Add basic clean architecture folders

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 ### :memo: Implementation
 
+<<<<<<< HEAD
```
Comment (SferaDev) [inline .github/pull_request_template.md:7]:
```suggestion
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -14,3 +15,8 @@
 
 -   [ ] Any change in the [API repo](https://github.com/EyeSeeTea/d2-api)? If so, what branch/PR?
 -   [ ] Any change in the [GUI library](https://github.com/EyeSeeTea/d2-ui-components)? If so, what branch/PR?
+=======
+### :art: Screenshots
+
+### :fire: Notes to the tester
+>>>>>>> 89ff4e9 (Add github template)
```
Comment (SferaDev) [inline .github/pull_request_template.md:22]:
```suggestion
```

File: .githooks/dep-check [hunk 3]
```diff
@@ -0,0 +1,8 @@
+function changed {
```
Comment (SferaDev) [inline .githooks/dep-check:1]: Missing bash shebang on top of githook


```suggestion
#!/bin/sh

function changed {
```

File: package.json [hunk 4]
```diff
@@ -21,12 +21,13 @@
         "@material-ui/core": "4.11.0",
         "@material-ui/icons": "4.9.1",
         "@material-ui/styles": "4.10.0",
+        "argparse": "^2.0.1",
```
Comment (SferaDev) [inline package.json:24]: Pin dependency?

File: package.json [hunk 5]
```diff
@@ -51,19 +52,22 @@
...
+            "pre-push": "yarn prettify && yarn lint && yarn update-po && yarn test",
+            "post-merge": "./.githooks/dep-check"
         }
     },
     "devDependencies": {
         "@babel/core": "7.11.4",
         "@babel/preset-typescript": "7.10.4",
         "@testing-library/jest-dom": "5.11.4",
         "@testing-library/react": "10.4.9",
+        "@types/argparse": "^2.0.2",
```
Comment (SferaDev) [inline package.json:70]: Pin dependency?

File: package.json [hunk 6]
```diff
@@ -96,6 +100,7 @@
         "prettier": "2.1.1",
         "react-test-renderer": "16.13.1",
         "ts-jest": "26.3.0",
+        "ts-node": "^9.0.0",
```
Comment (SferaDev) [inline package.json:103]: Pin dependency?

I believe sometime, somewhere in the future as if we were a big company, we might have a DevOps repo with a set of rules that execute as CI pipelines and ensure things like this (dependencies pinning, prettify, localize, domain/data mixed imports, etc...) and auto commits to the branch to fix them.

File: src/compositionRoot.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+import { Dhis2DataValueRepository } from "./data/Dhis2DataValueRepository";
+import { GetDataValuesUseCase } from "./domain/usecases/GetDataValuesUseCase";
+
+export function getCompositionRoot(api: D2Api) {
+    const dataValueRepository = new Dhis2DataValueRepository(api);
+
+    return {
+        dataValues: {
+            get: new GetDataValuesUseCase(dataValueRepository),
+        },
```
Comment (SferaDev) [inline src/compositionRoot.ts:11]: As opposed to our general class approach on composition root, we're not memorizing the Use-case here, nothing to worry about for skeleton though.

**Review by SferaDev — APPROVED** [review]
LGTM, minor comments added

File: config/testSetup.ts [hunk 8]
```diff
@@ -1,5 +1,7 @@
...
-import "@testing-library/jest-dom/extend-expect";
-
-configure({ adapter: new Adapter() });
+global.console = {
+    error: console.error,
+    info: console.info,
+    log: jest.fn(),
+    warn: jest.fn(),
+    debug: jest.fn(),
+};
```
Comment (SferaDev) [inline config/testSetup.ts:7]: import { TextDecoder, TextEncoder } from 'util';

```suggestion
global.console = {
    error: console.error,
    info: console.info,
    log: jest.fn(),
    warn: jest.fn(),
    debug: jest.fn(),
};
```

// Polyfill for encoding which isn't present globally in jsdom
global.TextEncoder = TextEncoder
global.TextDecoder = TextDecoder

File: src/compositionRoot.ts (hunk 7 above)
Comment (tokland) [reply src/compositionRoot.ts:11]: Do we need memoizing? It's an static object created when `getCompositionRoot` is called (at app start-up)

**Comment by tokland** [comment]
@SferaDev Changes applied, thx!

File: src/compositionRoot.ts (hunk 7 above)
Comment (SferaDev) [reply src/compositionRoot.ts:11]: That's right, nvm

**Pull request merged**